components/color-theme/element.js

```diff
@@ -58,63 +54,67 @@ export class MDNColorTheme extends L10nMixin(LitElement) {
       }
     }
   }
-  _toggleDropDown() {
-    // @ts-expect-error
-    const button = this.shadowRoot.querySelector(".dropdown");
-    // @ts-expect-error
-    const isExpanded = button.getAttribute("aria-expanded") === "true";
-    // @ts-expect-error
-    const dropdownId = button.getAttribute("aria-controls");
-    // @ts-expect-error
-    const dropdown = this.shadowRoot.querySelector(`#${dropdownId}`);
 
-    // @ts-expect-error
-    button.setAttribute("aria-expanded", !isExpanded);
-    // @ts-expect-error
-    dropdown.toggleAttribute("hidden");
+  _toggleDropDown() {
+    this._dropdown = !this._dropdown;
   }
-  _themeChanged() {}
 
   render() {
-    const setDefault = () => this._setMode("osDefault");
-    const setLight = () => this._setMode("light");
-    const setDark = () => this._setMode("dark");
-
     return html`<div class="color-theme">
       <button
         class="color-theme__button dropdown"
-        aria-expanded="false"
+        aria-expanded=${this._dropdown}
         aria-controls="color-theme__dropdown-1"
         @click=${this._toggleDropDown}
       >
-        ${this._getCurrent()} ${this.l10n`Theme`}
+        ${this._icon} ${this.l10n`Theme`}
       </button>
       <div
         class="color-theme__dropdown"
         id="color-theme__dropdown-1"
         data-side="right"
-        hidden
+        ?hidden=${!this._dropdown}
       >
         <ul class="color-theme__list">
           <li>
-            <button class="color-theme__option" @click=${setDefault}>
+            <button
+              class="color-theme__option"
+              data-mode="osDefault"
```

Comment:
```suggestion
              data-mode="light dark"
```

Since we already use `light` and `dark` values when forcing the theme, we could also skip the variable and use it here. It would simplify the logic in the `utils.js` a bit.